src/stk/_internal/topology_graphs/polymer/ncore/ncore.py

```diff
@@ -0,0 +1,551 @@
+                arm_building_blocks=tuple(arm_building_blocks),
+                vertices=vertices,
+            ),
+            edges=edges,
+            reaction_factory=reaction_factory,
+            construction_stages=(),
+            optimizer=optimizer,
+            num_processes=num_processes,
+        )
+
+    @staticmethod
+    def _get_vertices_and_edges(num_arms) -> "_VerticesAndEdges":
+        """
+        Get the vertices and edges of the topology graph.
+
+        Parameters:
+
+            num_arms:
+                The number of arms to add to the core.
+
+        Returns:
+
+            The vertices and edges of the topology graph.
+
+        """
+
+        if num_arms == 1:
+            arm_positions = [np.array((1, 0, 0))]
+        elif num_arms == 2:
+            arm_positions = [np.array((1, 0, 0)), np.array((-1, 0, 0))]
+        else:
+            thetas = np.linspace(0, 2 * np.pi, num_arms + 1)[:-1]
+            x_points = np.cos(thetas)
+            y_points = np.sin(thetas)
+            arm_positions = [
+                np.array((x, y, 0)) for x, y in zip(x_points, y_points)
+            ]
+
+        vertices: list[Vertex] = [CoreVertex(0, (0, 0, 0))]
+        edges: list[Edge] = []
+        for i, pos in enumerate(arm_positions):
+            vertices.append(SubstituentVertex(i + 1, pos))
+            edges.append(Edge(len(edges), vertices[0], vertices[-1]))
+
+        return _VerticesAndEdges(
+            vertices=tuple(vertices),
+            edges=tuple(edges),
+        )
+
+    def clone(self) -> typing.Self:
+        clone = self._clone()
+        clone._repr = self._repr
+        clone._repeating_unit = self._repeating_unit
+        clone._num_repeating_units = self._num_repeating_units
+        return clone
+
+    @staticmethod
+    def _normalize_repeating_unit(
+        repeating_unit: typing.Union[str, tuple[int, ...]],
+    ) -> tuple[int, ...]:
+        if isinstance(repeating_unit, tuple):
+            return repeating_unit
+
+        base = ord("A")
+        return tuple(ord(letter) - base for letter in repeating_unit)
+
+    def _get_building_block_vertices(
+        self,
+        core_building_block: BuildingBlock,
+        arm_building_blocks: tuple[BuildingBlock, ...],
+        vertices: tuple[Vertex, ...],
+    ) -> dict[BuildingBlock, abc.Sequence[Vertex]]:
+        building_block_vertices: dict[BuildingBlock, list[Vertex]] = {}
+        building_block_vertices[core_building_block] = [vertices[0]]
+
+        bb_order = self._repeating_unit * self._num_repeating_units
+        for bb_index, vertex in zip(bb_order, vertices[1:]):
+            bb = arm_building_blocks[bb_index]
+            building_block_vertices[bb] = building_block_vertices.get(bb, [])
+            building_block_vertices[bb].append(vertex)
+
+        # Have to do this to match the typing in the return statement
+        # (Sequence), suggestions for improvement would be great.
```

Comment:
yeah this is just a mypy issue afaik -- its made me sad in the past 

but the tests are passing -- so yyou're not getting an error?

Reply:
No erroes, so I assume it is fine. Annoying for aesthetics though